### PR TITLE
Allow custom lb IP and labels annotations in gen service

### PR DESCRIPTION
**Pull request by cezarsa**

### CODE REVIEW

File: pkg/apis/nginx/v1alpha1/nginx_types.go [hunk 1]
```diff
@@ -59,20 +59,27 @@ type Nginx struct {
...
 	Name string `json:"name"`
+}
+
+type NginxService struct {
 	// Type is the type of the service
 	Type string `json:"type"`
-	// ServiceIP is the IP of the service
-	ServiceIP string `json:"serviceIP"`
+	// LoadBalancerIP is an optional load balancer IP for the service
+	LoadBalancerIP string `json:"serviceIP"`
```
Comment (nettoclaudio) [inline pkg/apis/nginx/v1alpha1/nginx_types.go]: The JSON annotation should be `loadbalancerIP`, right?

```suggestion
	LoadBalancerIP string `json:"loadbalancerIP"`
```

File: pkg/k8s/k8s.go [hunk 2]
```diff
@@ -133,8 +151,9 @@ func NewService(n *v1alpha1.Nginx) *corev1.Service {
 					Port:       int32(80),
 				},
 			},
-			Selector: LabelsForNginx(n.Name),
-			Type:     nginxService(n),
+			Selector:       LabelsForNginx(n.Name),
+			LoadBalancerIP: lbIP,
```
Comment (nettoclaudio) [inline pkg/k8s/k8s.go]: Interesting... This way we can tie an external (outside of cluster) IP address to the Service, right?!

File: pkg/apis/nginx/v1alpha1/nginx_types.go (hunk 1 above)
Comment (cezarsa) [reply pkg/apis/nginx/v1alpha1/nginx_types.go]: Ouch.

File: pkg/k8s/k8s.go (hunk 2 above)
Comment (cezarsa) [reply pkg/k8s/k8s.go]: Yep, it depends on the cloud-provider supporting it though:
```
$ k explain svc.spec.loadBalancerIP
KIND:     Service
VERSION:  v1

FIELD:    loadBalancerIP <string>

DESCRIPTION:
     Only applies to Service Type: LoadBalancer LoadBalancer will get created
     with the IP specified in this field. This feature depends on whether the
     underlying cloud-provider supports specifying the loadBalancerIP when a
     load balancer is created. This field will be ignored if the cloud-provider
     does not support the feature.
```

Comment (nettoclaudio) [reply pkg/k8s/k8s.go]: Thanks.

**Pull request merged**